### PR TITLE
Ogre 2.1 port

### PR DESCRIPTION
**Pull request by rpaciorek**
- plugin port to Ogre 2.1 (with `#if OGRE_VERSION...` so it should not break 1.x support).
- changes in cmake build system to support Ogre 2.1 (add FindOGRE3D.cmake and option to disable builds demos)

### CODE REVIEW

File: src/OgreVideoManager.cpp [hunk 1]
```diff
@@ -105,16 +111,27 @@ namespace Ogre
 		TheoraVideoClip* clip=createVideoClip(new OgreTheoraDataStream(mInputFileName,group_name),TH_RGBA,0,1);
 		int w=nextPow2(clip->getWidth()),h=nextPow2(clip->getHeight());
 
-		TexturePtr t = TextureManager::getSingleton().createManual(name,group_name,TEX_TYPE_2D,w,h,1,0,PF_BYTE_RGBA,TU_DYNAMIC_WRITE_ONLY);
+		TexturePtr t = TextureManager::getSingleton().createManual(name,group_name,TEX_TYPE_2D,w,h,1,0,PF_X8R8G8B8,TU_DYNAMIC_WRITE_ONLY);
```
Comment (paroj) [inline src/OgreVideoManager.cpp]: please revert this unrelated change

File: src/OgreVideoManager.cpp [hunk 2]
```diff
@@ -105,16 +111,27 @@ namespace Ogre
...
 		
 		if (t->getFormat() != PF_X8R8G8B8) logMessage("ERROR: Pixel format is not X8R8G8B8 which is what was requested!");
 		// clear it to black
 
+#if OGRE_VERSION_MAJOR >= 2 && OGRE_VERSION_MINOR >= 1
+		unsigned char* texData=(unsigned char*) t->getBuffer()->lock(v1::HardwareBuffer::HBL_DISCARD);
```
Comment (paroj) [inline src/OgreVideoManager.cpp]: rather than duplicating the whole line, add a `using v1::HardwareBuffer;` at the top of the file.

File: src/OgreVideoManager.cpp [hunk 3]
```diff
@@ -156,7 +173,11 @@ namespace Ogre
 				int w=f->getStride(),h=f->getHeight();
 				TexturePtr t=mTextures[(*it)->getName()];
 
+#if OGRE_VERSION_MAJOR >= 2 && OGRE_VERSION_MINOR >= 1
+				unsigned char *texData=(unsigned char*) t->getBuffer()->lock(v1::HardwareBuffer::HBL_DISCARD);
```
Comment (paroj) [inline src/OgreVideoManager.cpp]: same as before

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,8 +1,10 @@
 cmake_minimum_required (VERSION 2.8)
 project(OgreVideo)
 
-find_package(OGRE 1.11 REQUIRED COMPONENTS Bites)
+set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
```
Comment (paroj) [inline CMakeLists.txt]: do not modify the module path. Instead put the FindOGRE.cmake for 2.1 in one of the predefined search locations:
https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
+option(BUILD_DEMOS "Build demos (required Ogre Bites)" ON)
 
+find_package(OGRE3D)
```
Comment (paroj) [inline CMakeLists.txt]: remove the custom (copypasted) find script. This gets out of sync with upstream rapidly. Instead make sure that the upstream version is copied to a system wide location.

File: CMakeLists.txt [hunk 6]
```diff
@@ -22,10 +24,14 @@ target_include_directories(Plugin_TheoraVideoSystem PUBLIC include/)
 target_link_libraries(Plugin_TheoraVideoSystem PUBLIC theoraplayer OgreMain)
 set_target_properties(Plugin_TheoraVideoSystem PROPERTIES PREFIX "")
 
-add_executable(player demos/player/player.cpp)
-target_link_libraries(player OgreBites Plugin_TheoraVideoSystem)
-
-file(COPY demos/resources.cfg DESTINATION ${CMAKE_BINARY_DIR})
+if (BUILD_DEMOS)
+	find_package(OGRE REQUIRED COMPONENTS Bites)
```
Comment (paroj) [inline CMakeLists.txt]: instead of searching a second time here, remove the required part above, but check `OGRE_Bites_FOUND ` here. see: https://ogrecave.github.io/ogre/api/latest/setup.html

File: src/OgreVideoManager.cpp (hunk 1 above)
Comment (rpaciorek) [reply src/OgreVideoManager.cpp]: OK, but in next line should still compare `getFormat()` with `PF_X8R8G8B8`?

Comment (paroj) [reply src/OgreVideoManager.cpp]: no, it should be `PF_BYTE_RGBA` as well.
